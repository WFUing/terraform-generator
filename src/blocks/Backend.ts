import { Argument, Attribute } from '../arguments';
import { TerraformArgs, Util } from '../utils';
import { Block } from '.';

/**
 * @category Block
 */
export class Backend extends Block {

  readonly type: string;

  /**
   * Construct backend.
   *
   * Refer to Terraform documentation on what can be put as type & arguments.
   *
   * @param type type
   * @param args arguments
   */
  constructor(type: string, args: TerraformArgs) {
    super('backend', [type], args, undefined, true);

    this.type = type;
  }

  override asArgument(): Argument {
    return new Argument(JSON.stringify(this.type));
  }

  override attr(_name: string): Attribute {
    throw Util.inaccessibleMethod();
  }

}
