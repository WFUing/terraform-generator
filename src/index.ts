export { BlockArgs } from './utils';

export { TerraformGenerator, WriteOptions } from './TerraformGenerator';

export {
  Block,
  Backend,
  Comment,
  Data,
  Import, ImportArgs,
  Locals,
  Module, ModuleArgs,
  Output, OutputArgs,
  Provider,
  Provisioner, ProvisionerArgs, ProvisionerType,
  Resource, ResourceToDataOptions,
  Variable, VariableArgs
} from './blocks';

export {
  Argument, arg,
  Attribute, attr,
  Function, fn,
  Heredoc, heredoc
} from './arguments';

export {
  List, list,
  Map, map
} from './types';
