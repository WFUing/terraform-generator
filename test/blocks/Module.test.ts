import { arg4 } from '..';
import { Module } from '../../src/blocks';

test('Module', () => {
  const module = new Module('name', {
    source: 'source',
    version: '1',
    ...arg4
  });
  expect(module.toTerraform()).toMatchSnapshot();
  expect(module.asArgument().toTerraform()).toMatchSnapshot();
  expect(module.attr('attr').toTerraform()).toMatchSnapshot();
});
