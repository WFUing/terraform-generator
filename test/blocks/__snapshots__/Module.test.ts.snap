// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Module 1`] = `
"module &tfgquot;name&tfgquot;{
source = &tfgquot;source&tfgquot;
version = &tfgquot;1&tfgquot;
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg102 = {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
}
arg103 = [
{
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
},
{
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
},
{
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
arg101 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
arg100 {
arg0 = &tfgquot;a&tfgquot;
arg1 = [
&tfgquot;b&tfgquot;,
&tfgquot;c&tfgquot;,
&tfgquot;d&tfgquot;
]
arg2 = 0
arg3 = [
1,
2,
3
]
arg4 = true
arg5 = [
false,
true,
false
]
arg6 = type.name.attr
arg7 = [
type.name.attr,
type.name.attr,
type.name.attr
]
arg8 = type.name
arg9 = [
type.name,
type.name,
type.name
]
arg10 = [
&tfgquot;e&tfgquot;,
4,
true,
type.name.attr,
type.name
]
arg11 = arg
arg12 = <<EOT
heredoc
EOT

arg13 = fn(1, 2, 3)
arg14 = fn(type.name.attr, type.name.attr, type.name.attr)
}
}
}
]
}

"
`;

exports[`Module 2`] = `"module.name"`;

exports[`Module 3`] = `"module.name.attr"`;
